pex/environment.py

```diff
@@ -502,14 +515,21 @@ def resolve(self):
         # type: () -> Iterable[Distribution]
         if self._resolved_dists is None:
             all_reqs = [Requirement.parse(req) for req in self._pex_info.requirements]
+            exclude_configuration = ExcludeConfiguration.create(excluded=self._pex_info.excluded)
             self._resolved_dists = tuple(
                 fingerprinted_distribution.distribution
-                for fingerprinted_distribution in self.resolve_dists(all_reqs)
+                for fingerprinted_distribution in self.resolve_dists(
+                    all_reqs, exclude_configuration=exclude_configuration
+                )
             )
         return self._resolved_dists
 
-    def resolve_dists(self, reqs):
-        # type: (Iterable[Requirement]) -> Iterable[FingerprintedDistribution]
+    def resolve_dists(
+        self,
+        reqs,  # type: Iterable[Requirement]
+        exclude_configuration=ExcludeConfiguration(),  # type: ExcludeConfiguration
```

Comment:
`pex/resolve/pex_repository_resolver.py` also calls this function; should it propagate that pex's exclusions?

Reply:
No. That use case flow is part of the 1-2 (it's one of the 3 ways step 2 is implemented):
1. add repository pex dists to dep manager
2. add resolved dists to dep manager
3. now that the full set of dists is known, can the proper excludes be calculated and no earlier.